napari/components/viewer_model.py

```diff
@@ -39,47 +44,32 @@ class ViewerModel(AddLayersMixin, KeymapHandler, KeymapProvider):
         Preset color palettes.
     """
 
-    themes = palettes
-
-    def __init__(
-        self, title='napari', ndisplay=2, order=None, axis_labels=None
-    ):
-        super().__init__()
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            status=Event,
-            help=Event,
-            title=Event,
-            interactive=Event,
-            cursor=Event,
-            reset_view=Event,
-            active_layer=Event,
-            palette=Event,
-            grid=Event,
-            layers_change=Event,
-        )
-
+    title: str = 'napari'
+    status: str = 'Ready'
+    help: str = ''
+    cursor: str = 'standard'
+    cursor_size: Optional[int] = None
+    interactive: bool = True
+    active_layer: Optional[int] = None
+    grid_size: Tuple[int, int] = (1, 1)
+    theme: str = DEFAULT_THEME
+
+    grid_stride: ClassVar[int] = 1
+    palette: ClassVar[Dict[str, str]] = palettes[DEFAULT_THEME]
+    layers: ClassVar[LayerList] = []
+
+    ndisplay: InitVar[int] = 2  # type: ignore
+    order: InitVar[Optional[Tuple[int, ...]]] = None  # type: ignore
+    axis_labels: InitVar[Optional[Sequence[str]]] = None  # type: ignore
+
+    def __post_init__(self, ndisplay, order, axis_labels):
+        super().__init__()  # will this work?
+        self.events.add(reset_view=None, grid=None, layers_change=None)
+        self.layers = LayerList()
```

Comment:
Is it not possible to do the `LayerList()` call in the actual init? i.e. `layers: ClassVar[LayerList] = LayerList()`?

Reply:
updated, 
```python
layers: LayerList = field(default_factory=LayerList, metadata={'events': False})
```
and see comment above about mutable defaults: